### PR TITLE
[stable8.2] allow putting the part file in the view root

### PR DESCRIPTION
**Pull request by icewind1991**
Backport of #22684

cc @PVince81

### CODE REVIEW

**Comment by PVince81** [comment]
@icewind1991 The config.sample.php piece is missing

**Comment by PVince81** [comment]
Looks like I destroyed it in my force push on master?

**Comment by PVince81** [comment]
Tested, works :+1: (with sabotaged `rename` method in SMB)

Second review @nickvergessen @schiesbn ?

**Comment by PVince81** [comment]
or @MorrisJobke

**Comment by nickvergessen** [comment]
:+1: as per original PR

**Comment by PVince81** [comment]
:-1: doesn't work properly when the root user has a quota.

We might need to backport the fix that disables the quota wrapper for part files.

**Comment by PVince81** [comment]
Requires d32fc2d7718db89e30c636889ebed522db82e5cc to make this work.

Pushing...

**Comment by icewind1991** [comment]
:+1: for @PVince81's backport part

**Comment by PVince81** [comment]
With that commit it works fine now, the part file is no longer truncated :+1:

**Comment by PVince81** [comment]
Let me run one last smashbox with the setting enabled, just in case, then we can merge this. (well, after the master one is merged too)

**Comment by PVince81** [comment]
Damn... I ran the smashbox tests with the option disabled.

Running now with `'part_file_in_storage' => false`

**Comment by PVince81** [comment]
Tests passed with the new flag, all good!

I'll set to "To release" so @DeepDiver1975 will hopefully merge it after merging the master one.

File: apps/dav/tests/unit/connector/sabre/requesttest/partfileinrootupload.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?php
```
Comment (MorrisJobke) [inline apps/dav/tests/unit/connector/sabre/requesttest/partfileinrootupload.php]: I guess this test was backported wrongly :see_no_evil:

**Pull request merged**